File: src/custom_robot_controller.py
import rospy
from sensor_msgs.msg import LaserScan
from geometry_msgs.msg import Twist

class csvExporterNode():


    def process_LaserScan(self,data):

        twist_msg = Twist()

        #########################################
        #
        # Write your code here
        #
        # The values from the LiDAR are in data.ranges
        # For more information on the data contained in a LaserScan message, consult http://docs.ros.org/en/api/sensor_msgs/html/msg/LaserScan.html
        #
        # The following are used to set the linear and angular velocities of the robot
        # twist_msg.linear.x = 0;
        # twist_msg.angular.z = 0;
        ########################################

        self.twist_pub.publish(twist_msg)
        


    def __init__(self):

        rospy.init_node('custom_robot_controller', anonymous=True)

        self.scan_topic = "/simple_lidar"   # uncomment only one of these lines at a time, depending on which LiDAR you wish to use
        #self.scan_topic = "/static_laser"   # "/static_laser" uses the complete LiDAR scan, "/simple_lidar" uses a simplified version

        self.twist_topic = "/cmd_vel"

        rospy.Subscriber(
                self.scan_topic,
                LaserScan,
                self.process_LaserScan,
                buff_size=10000000,
                queue_size=1)

        self.twist_pub = rospy.Publisher(self.twist_topic, Twist, queue_size=1)




if __name__ == '__main__':
    try:
        csvExpNode = csvExporterNode()
        
        rospy.spin()        

    except rospy.ROSInterruptException:
        pass

File: src/custom_robot_controller.cpp
#include <ros/ros.h>
#include <geometry_msgs/Twist.h>
#include <sensor_msgs/LaserScan.h>


class CustomRobotController
{
    public:
        CustomRobotController();

    private:
        void processLaserScan(const sensor_msgs::LaserScan &lidar_scan_msg);
        ros::NodeHandle nh;
        std::string scan_topic, twist_topic;
        ros::Subscriber scan_sub;
        ros::Publisher twist_pub;

};

CustomRobotController::CustomRobotController():
    scan_topic("static_laser"), //"static_laser" uses the complete lidar scan, "simple_lidar" uses a simplified version
    twist_topic("cmd_vel")
{
    scan_sub = nh.subscribe(scan_topic, 1, &CustomRobotController::processLaserScan, this);
    twist_pub = nh.advertise<geometry_msgs::Twist>(twist_topic, 1);
}

void CustomRobotController::processLaserScan(const sensor_msgs::LaserScan &lidar_scan_msg)
{

    geometry_msgs::Twist twist_msg;

    /**
        Write your code here

        The values from the LiDAR are in lidar_scan_msg.ranges
        For more information on the data contained in a LaserScan message, consult http://docs.ros.org/en/api/sensor_msgs/html/msg/LaserScan.html

        The following are used to set the linear and angular velocities of the robot
        twist.linear.x = 0;
        twist.angular.z = 0;
     */

    twist_pub.publish(twist_msg);

}

int main(int argc, char** argv)
{
  ros::init(argc, argv, "custom_robot_controller");
  CustomRobotController robot_controller;

  ros::spin();
  
  return(0);
}
